src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -11,103 +11,80 @@
-
-    // Gets the actual forward/backward motion of the robot.
-    double moveRate = getMoveRate();
-    // Gets the acutal speed difference between left and right sides of the robot.
-    double turnRate = getTurnRate();
-
-    /**
-     * Generates a motor power value that will move the robot where the driver wants
-     * it to go, based on the difference between the current motor rate and the
-     * target.
-     */
-    double move = pidEnabled ? movePID.calculate(moveRate, moveRequest) : moveRequest;
-    double turn = pidEnabled ? turnPID.calculate(turnRate, turnRequest) : turnRequest;
-
-    // Forces the speed limit on the output of the PID controllers.
-    double m_r = Utils.clip(move, -speedLimiter, speedLimiter);
-    double t_r = Utils.clip(turn, -speedLimiter, speedLimiter);
-
-    // Sends the actual drive command that activates the motors.
-    drive.arcadeDrive(m_r, t_r, false);
-    // }
+  public void drive(double moveRequest, double turnRequest) {
+    var chassisSpeeds = new ChassisSpeeds(moveRequest, 0, turnRequest);
+    DifferentialDriveWheelSpeeds wheelSpeeds = kinematics.toWheelSpeeds(chassisSpeeds);
+    double leftMPerS = wheelSpeeds.leftMetersPerSecond;
+    double rightMPerS = wheelSpeeds.rightMetersPerSecond;
+    double leftETPerS = FeetToEncTicks(
+        Units.metersToFeet(leftMPerS) < 12 && Units.metersToFeet(leftMPerS) > -12 ? Units.metersToFeet(leftMPerS) : 12);
+    double rightETPerS = FeetToEncTicks(
+        Units.metersToFeet(rightMPerS) < 12 && Units.metersToFeet(rightMPerS) > -12 ? Units.metersToFeet(leftMPerS)
+            : 12);
+    leftFront.set(ControlMode.Velocity, leftETPerS);
+    rightFront.set(ControlMode.Velocity, rightETPerS);
   }
 
   public void resetEncoders() {
     leftEnc.setQuadraturePosition(0, 0); // Resets both encoders so they count from zero.
     rightEnc.setQuadraturePosition(0, 0);
   }
 
-  public double getMoveRate() {
-    return (leftEnc.getQuadratureVelocity() + rightEnc.getQuadratureVelocity()) / 2; // Averages the speed of both drive
-                                                                                     // sides to get mean forward
-                                                                                     // velocity.
-  }
-
-  public double getTurnRate() {
-    return (leftEnc.getQuadratureVelocity() - rightEnc.getQuadratureVelocity()); // Subtracts the speed of both drive
-                                                                                 // sides to get net turn vector.
-  }
+  /**
+   * Averages the speed of both drive sides to get mean forward velocity
+   */
 
   public void stop() {
-    drive.arcadeDrive(0, 0, false);
+    leftFront.stopMotor();
+    rightFront.stopMotor();
   }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
   }
 
-  public void useOutput(double output, double setpoint) {
-
-  };
+  private double FeetToEncTicks(double ft) {
```

Comment:
Java conventions dictate that class names start with an upper case letter and class methods start with a lower case letter.